File: Daily/442. Find All Duplicates in an Array/Solution.cpp
#include <iostream>
#include <vector>
#include <functional>
#include <unordered_map>
using namespace std;
class Solution {
    public:
        vector<int> findDuplicates(vector<int>& nums) {
            vector<int> duplicates; unordered_map<int,int> count_Freq;
            for(const int& x: nums) count_Freq[x]++;
            unordered_map<int,int>::iterator itr;
            for(itr = count_Freq.begin(); itr != count_Freq.end(); itr++)
                if(itr->second == 2) duplicates.emplace_back(itr->first);
            return duplicates;
        }
};
int main() {
    function<void(vector<int>)> printArray = [&](vector<int> array) {
        for(int index = 0; index < array.size(); ++index) {
            if(index == 0) cout << '[';
            cout << array[index];
            if(index == array.size()-1) cout << ']';
            else cout << ',';
        }
        cout << endl;
    };
    int n; cin >> n; vector<int> nums(n, 0); Solution sol;
    for(int index = 0; index < n; ++index) cin >> nums[index];
    cout << "Original Array: "; printArray(nums);
    vector<int> array = sol.findDuplicates(nums);
    cout << "Duplicates in Array: "; printArray(array);
}

File: Daily/442. Find All Duplicates in an Array/Solution.py
from typing import List
from collections import Counter
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        '''
            duplicates, count_Freq = [], {}
            for num in nums: count_Freq[num] = count_Freq.get(num, 0) + 1
            for num, freq in count_Freq.items():
                if freq == 2: duplicates.append(num)
            return duplicates
        '''
        return [num for num, freq in Counter(nums).items() if freq == 2]
if __name__ == '__main__':
    nums = [int(x) for x in input().split()]; sol = Solution()
    print('Original Array:', nums)
    print('Duplicates in Array:', sol.findDuplicates(nums))